### ISSUE TITLE
Master configuration "pillar_merge_lists" has no effect

### ISSUE THREAD

**Issue by anlutro** [saltstack/salt#30809]
In 2015.8.4, "pillar_merge_lists" was introduced. It was erroneously set as the default behaviour until 2015.8.7, but even now, you can't seem to change it in the master configuration - changing it has no effect.

Old issue text:

Using salt-ssh, checked out on the tag `v2015.8.3`, lists are overwritten, not merged, in pillars.

Checked out on the tag `v2015.8.4`, lists are merged. This seems to be the case in the latest 2015.8 HEAD as well.

It doesn't even seem to matter what I set `pillar_merge_lists` to in my config file.

I see several additions like these in the code between 2015.8.3 and 2015.8.4:

```
self.opts.get('pillar_merge_lists', 'False')

merge_lists = __opts__.get(
    'pillar_merge_lists',
    'False'
)
```

Why is False inside quotes? The string `'False'` evaluates to true, which is probably the cause of this bug.

Related PRs: https://github.com/saltstack/salt/pull/30062 https://github.com/saltstack/salt/pull/30458

**Comment by anlutro** [comment]
This seems to be an issue in the lastest 2015.8 HEAD as well.

**Comment by anlutro** [comment]
I see several additions like these in the code between 2015.8.3 and 2015.8:

```
self.opts.get('pillar_merge_lists', 'False')

merge_lists = __opts__.get(
    'pillar_merge_lists',
    'False'
)
```

Why on earth is False inside quotes? The string `'False'` evaluates to true, which is probably the cause of this bug.

**Comment by anlutro** [comment]
Related PRs: https://github.com/saltstack/salt/pull/30062 https://github.com/saltstack/salt/pull/30458

**Comment by anlutro** [comment]
Removing the quotes around `False` changes the behavior back to the default in 2015.8.3, but setting it to `true` in the master config now doesn't seem to have any effect.

**Comment by jfindlay** [comment]
@anlutro, thanks for fixing this.

**Comment by anlutro** [comment]
@jfindlay it still looks like setting `pillar_merge_lists` in the master config has no effect, should I open a new issue for that or can you just re-open this one?

**Comment by puneetk** [comment]
made a minor change for how pillar.get uses the master config. Not sure why the builds are failing, i didnt change the tornado code

**Comment by jfindlay** [comment]
@puneetk, that is a test that fails intermittently, that we haven't tracked down yet.  I reran the tests and they all passed.

**Comment by puneetk** [comment]
Sweet, thanks @jfindlay . Anything more you need from me to merge this in ?

**Comment by jfindlay** [comment]
@puneetk, it seems fine to me.

**Comment by rallytime** [comment]
@anlutro Does the fix in #32665 fix this issue for you? I am thinking if it does, we should backport that fix to the 2015.8 branch. Thoughts?
